packages/db/src/lib/retry.ts

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+import { Knex } from 'knex';
+import pRetry from 'p-retry';
+import Logger from '@cumulus/logger';
+
+type PromiseReturnType<T> = T extends (
+  method: Promise<any> | Knex.QueryBuilder
+) => infer R ? Promise<R> : never;
+
+export const RetryOnDbConnectionTerminateError = async <T>(
+  method: Promise<any> | Knex.QueryBuilder,
```

Comment:
Does `Promise<any>` includes `Knex.QueryBuilder`?  Remove `Knex.QueryBuilder` if we can't define all specific types.

Reply:
`Promise<any>` doesn't include `Knex.QueryBuilder` unfortunately 😕